### ISSUE TITLE
Booking information not updated for State Tracker[BUG] 

### ISSUE THREAD

**Issue by ChrisGeishauser** [thu-coai/ConvLab-2#175]
Hi all,

in the script ConvLab-2/convlab2/policy/vector/vector_multiwoz.py in line 168 onwards, there is the following code:       

 book = np.zeros(len(self.db_domains))
        for i, domain in enumerate(self.db_domains):
            if state['belief_state'][domain.lower()]['book']['booked']:
                book[i] = 1.

This checks whether an entity has been booked by the system for every domain. Yet I can not find that the belief_state is updated after an entity has been booked.

I would expect that after an entity has been booked, the belief_state will be updated. Is that missing or did I overlook it?

Thanks in advance!

**Comment by liangrz15** [comment]
In the script ConvLab-2/convlab2/policy/vector/vector_multiwoz.py, the method `state_vectorize(self, state)` is used to **create the vector version of the given state**. It does not try to update anything.

Currently, the output `state_vec` consists of six parts (for readability), two of which are `belief_state` and `book`.  Both `belief_state` and `book` are created according to the input `state`.

The input `state` of `state_vectorize(self, state)` is created in **dst module**. If you are interested in how the book information in input `state` is updated, you can check ConvLab-2/convlab2/dst/rule/multiwoz/dst.py.

**Comment by ChrisGeishauser** [comment]
Thanks for your quick answer!

> In the script ConvLab-2/convlab2/policy/vector/vector_multiwoz.py, the method state_vectorize(self, state) is used to create the vector version of the given state. It does not try to update anything.

I just used it for checking if the case "if state['belief_state'][domain.lower()]['book']['booked']:" is ever triggered and it was not.

> The input state of state_vectorize(self, state) is created in dst module. If you are interested in how the book information in input state is updated, you can check ConvLab-2/convlab2/dst/rule/multiwoz/dst.py.

I also checked that but it seems it only uses the user-act to update it right? But the information if an entity has been booked should be updated using the system-act, right?

**Comment by zqwerty** [comment]
You are right. 'booked' is not updated by ruleDST. I think booked information should be updated by the policy module (sys-act).
